pkg/resources/dynamic_table.go

```diff
@@ -0,0 +1,342 @@
+		Type:        schema.TypeBool,
+		Description: "Whether auto-clustering is enabled on the dynamic table. Not currently supported for dynamic tables.",
+		Computed:    true,
+	},
+	"scheduling_state": {
+		Type:        schema.TypeString,
+		Description: "Displays RUNNING for dynamic tables that are actively scheduling refreshes and SUSPENDED for suspended dynamic tables.",
+		Computed:    true,
+	},
+	"last_suspended_on": {
+		Type:        schema.TypeString,
+		Description: "Timestamp of last suspension.",
+		Computed:    true,
+	},
+	"is_clone": {
+		Type:        schema.TypeBool,
+		Description: "TRUE if the dynamic table has been cloned, else FALSE.",
+		Computed:    true,
+	},
+	"is_replica": {
+		Type:        schema.TypeBool,
+		Description: "TRUE if the dynamic table is a replica. else FALSE.",
+		Computed:    true,
+	},
+	"data_timestamp": {
+		Type:        schema.TypeString,
+		Description: "Timestamp of the data in the base object(s) that is included in the dynamic table.",
+		Computed:    true,
+	},
+}
+
+// DynamicTable returns a pointer to the resource representing a dynamic table.
+func DynamicTable() *schema.Resource {
+	return &schema.Resource{
+		Create: CreateDynamicTable,
+		Read:   ReadDynamicTable,
+		Update: UpdateDynamicTable,
+		Delete: DeleteDynamicTable,
+
+		Schema: dynamicTableShema,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+	}
+}
+
+// ReadDynamicTable implements schema.ReadFunc.
+func ReadDynamicTable(d *schema.ResourceData, meta interface{}) error {
+	db := meta.(*sql.DB)
+	client := sdk.NewClientFromDB(db)
+
+	id := helpers.DecodeSnowflakeID(d.Id()).(sdk.SchemaObjectIdentifier)
+	dynamicTable, err := client.DynamicTables.ShowByID(context.Background(), id)
+	if err != nil {
+		log.Printf("[DEBUG] dynamic table (%s) not found", d.Id())
+		d.SetId("")
+		return nil
+	}
+	if err := d.Set("name", dynamicTable.Name); err != nil {
+		return err
+	}
+	if err := d.Set("database", dynamicTable.DatabaseName); err != nil {
+		return err
+	}
+	if err := d.Set("schema", dynamicTable.SchemaName); err != nil {
+		return err
+	}
+	if err := d.Set("warehouse", dynamicTable.Warehouse); err != nil {
+		return err
+	}
+	if err := d.Set("comment", dynamicTable.Comment); err != nil {
+		return err
+	}
+	tl := map[string]interface{}{}
+	if dynamicTable.TargetLag == "DOWNSTREAM" {
+		tl["downstream"] = true
+		if err := d.Set("target_lag", []interface{}{tl}); err != nil {
+			return err
+		}
+	} else {
+		tl["maximum_duration"] = dynamicTable.TargetLag
+		if err := d.Set("target_lag", []interface{}{tl}); err != nil {
+			return err
+		}
+	}
+	text := dynamicTable.Text
+	if strings.Contains(text, "OR REPLACE") {
+		if err := d.Set("or_replace", true); err != nil {
+			return err
+		}
+	} else {
+		if err := d.Set("or_replace", false); err != nil {
+			return err
+		}
+	}
+	// trim up to " ..AS" and remove whitespace
+	query := strings.TrimSpace(text[strings.Index(text, "AS")+3:])
```

Comment:
Hello. Turns out this will be incorrect when anything in the table description text contains the 'AS' letter sequence, either the DB name, schema name, table name, comment, or warehouse name. It will end up splitting at the incorrect location.